stardis/base.py

```diff
@@ -18,7 +18,7 @@
 schema = os.path.join(base_dir, "config_schema.yml")
 
 
-def run_stardis(config_fname, tracing_lambdas_or_nus):
+def run_stardis(config_fname, tracing_lambdas_or_nus, atom_data=None):
```

Comment:
Adding atom data as an argument is good, but I would also add the config as an argument. I think the easiest way of doing this is just moving lines 42 through 56 into a separate function that prepares both atom_data and the config file, so we have separate functions for reading in the outer data and running our simulation. That way in python you could call the read in function, make changes to either the atom data or some of the config options, and then run the simulation with the modified things. 

Lines 59 through 74 could also be a separate function, but that's lower priority and could happen after that first change.

Reply:
I expect the difficulty of this to more come from the other party of the code that this will likely break, in the tests and benchmarks and things. On second thought, maybe the less painful way of doing this is to create two functions that run_stardis() can call. There can be a prepare_inputs() function for the preparation, and then a run_simulation() function for the rest of run_stardis(). That way the functionality of run_stardis() itself won't change and the tests and benchmarks should still function in the short term.